src/utils/types.tsx

```diff
@@ -14,6 +14,12 @@ export type ComponentWithAs<Props, DefaultType extends As> = {
   (props: PropsWithAs<Props, DefaultType>): JSX.Element;
 };
 
+/**
+ * @template Props Component Props
+ * @template RefProp HTML intrinsic type, eg: HTMLDivElement
+ * @template DefaultType string, eg: "div"
+ * @param component
+ */
```

Comment:
I saw them in the twitter !!
Using `@template`

Reply:
Yeah.. its actually officially documented but never noticed..
https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html